RabbitMQ.xs

```diff
@@ -483,29 +484,14 @@ int internal_recv(HV *RETVAL, amqp_connection_state_t conn, int piggyback, int t
 
           // Handle kind UTF8 and kind BYTES
           case AMQP_FIELD_KIND_UTF8:
-            hvalue = newSVpvn(
-              header_entry->value.value.bytes.bytes,
-              header_entry->value.value.bytes.len
-            );
-            /* If it's UTF8, set the flag on... */
-            if (header_entry->value.kind == AMQP_FIELD_KIND_UTF8) {
-              SvUTF8_on(hvalue);
-            }
-            hv_store( headers,
-                header_entry->key.bytes, header_entry->key.len,
-                hvalue,
-                0
-            );
-            break;
-
           case AMQP_FIELD_KIND_BYTES:
-            hvalue = newSVpvn(
-              header_entry->value.value.bytes.bytes,
-              header_entry->value.value.bytes.len
-            );
             hv_store( headers,
                 header_entry->key.bytes, header_entry->key.len,
-                hvalue,
+                newSVpvn_utf8(
```

Comment:
Ok. Makes sense. Thanks for the answer. Had to dig into this macro a bit, wish they'd link to that stuff in the documentation.